### ISSUE TITLE
`wsl --update` and MSI require Admin rights

### ISSUE THREAD

**Issue by byjrack** [microsoft/WSL#11652]
### Windows Version

10.0.22621.3593

### WSL Version

2.0.0.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.123.1-1

### Distro Version

n/a

### Other Software

n/a

### Repro Steps

`wsl --update` fails with a 1603 (attempt to update to 2.1.5 gives a Wsl/UpdatePackage/E_ABORT)

Download of 2.1.5 from releases and install as MSI also requires Admin for all actions in the MSI.  Tried MSIINSTALLPERUSER but no change.

`winget install Microsoft.WSL` shows the VMP and OS feature requirement which are already installed, but the get-optfeature and capability cmdlets require elevation so status may be inaccurate.

### Expected Behavior

Updates would function without Admin

### Actual Behavior

Updates require Admin

### Diagnostic Logs

I am just trying to get a consistent approach to managing WSL in the environment.  It seemed like with it in Store and semi-recent updates it was happy to update as a User.  UAC requires some extra steps in my org so it pushes me into a bunch of extra activities and overhead to keep WSL up to date.  I get that WSL2 needs to play with some OS level features like Host Network Service so if maintaining WSL requires Admin I can work to that as the standard, but it seems like a moving target.

[wsl-install-logs.txt](https://github.com/user-attachments/files/15552478/wsl-install-logs.txt)

**Comment by OneBlue** [comment]
@byjrack: if you have the store installed, you should be able to update via the store once WSL is installed. 

You can use this link to access the WSL page: https://aka.ms/wslstorepage

**Comment by byjrack** [comment]
Yeah doesn't seem to work as Store isn't keeping it updated and if it requires admin from 2.0.0 forward that won't work.

https://apps.microsoft.com/detail/9p9tqf7mrm4r still exists which is what we had syndicated via Intune, but looks to not have been updated in a while.

**Comment by OneBlue** [comment]
The store app is up to date.  It's currently on the latest release (2.1.5).

**Comment by byjrack** [comment]
¯\_(ツ)_/¯

Maybe it's the lack of admin rights for users, but doesn't seem to be working for us or we are syndicating the wrong store id.

So is there a method to keep wsl updated without admin?  

If Store is that method I can dig into ime logs to see if there is something, but if it's not using msix model than given the user scope it would fail.  I know when I checked Winget as well I only got the Microsoft.wsl entry as well from Winget portal and not the Store entry.  That's why I figured the Store option was on ice.
